src/models/report.es6.js

```diff
@@ -3,47 +3,50 @@ import Base from './base';
 class Report extends Base {
   _type = 'Report';
 
-  constructor(props) {
-    super(props);
-
-    var report = this;
+  validators () {
+    const reason = this.reasonValidator.bind(this);
+    const otherReason = this.otherReasonValidator.bind(this);
+    const thingId = this.thingIdValidator.bind(this);
 
     this.validators = {
-      reason: function() {
-        if (this.get('other_reason') && !(this.get('reason') === 'other')) {
-          return false;
-        }
-
-        var reasonValid = Base.validators.minLength(this.get('reason'), 1) &&
-                          Base.validators.maxLength(this.get('reason'), 100);
-
-        var otherReasonValid = !this.get('other_reason') ||
-                               (
-                                 Base.validators.minLength(this.get('other_reason'), 1) &&
-                                 Base.validators.maxLength(this.get('other_reason'), 100)
-                               );
-
-        return reasonValid && otherReasonValid;
-      }.bind(report),
-
-      other_reason: function() {
-        if (this.get('reason') !== 'other') {
-          if (this.get('other_reason')) {
-            return false;
-          }
-        } else if (!this.get('reason')) {
-          return false;
-        }
-
-        return Base.validators.maxLength(this.get('thing_id'), 100);
-      }.bind(report),
-
-      thing_id: function() {
-        return Base.validators.minLength(this.get('thing_id'), 6) &&
-               Base.validators.maxLength(this.get('thing_id'), 10);
-      }.bind(report),
+      reason,
+      otherReason,
+      thingId,
     };
   }
-};
+
+  reasonValidator () {
+    if (this.get('other_reason') && !(this.get('reason') === 'other')) {
+      return false;
+    }
+
+    const reasonValid = Base.validators.minLength(this.get('reason'), 1) &&
```

Comment:
:no_bell: A validator for `lengthInRange` would be nice to add in the future if we have more of these range checks